packages/devtools_app/lib/src/table.dart

```diff
@@ -117,18 +119,36 @@ class FlatTableState<T> extends State<FlatTable<T>>
     sortData(widget.sortColumn, widget.sortDirection);
   }
 
-  List<double> _computeColumnWidths(double maxWidth) {
+  @visibleForTesting
+  List<double> computeColumnWidths(double maxWidth) {
+    // Subtract width from outer padding around table.
     maxWidth -= 2 * defaultSpacing;
+    maxWidth -= widget.numSpacers * defaultSpacing;
+    maxWidth = max(0, maxWidth);
 
     final unconstrainedCount =
         widget.columns.where((col) => col.fixedWidthPx == null).length;
-    final allocatedWidth = widget.columns.fold(0,
-        (val, col) => col.fixedWidthPx == null ? val : col.fixedWidthPx + val);
-    final available = maxWidth - allocatedWidth;
+    double allocatedWidth = 0;
+    for (var col in widget.columns) {
+      if (col.fixedWidthPx != null) {
+        allocatedWidth += col.fixedWidthPx;
+      } else {
+        if (col.minWidthPx != null) {
+          allocatedWidth += col.minWidthPx;
+        }
+      }
+    }
+    final available = max(0, maxWidth - allocatedWidth);
 
     final widths = <double>[];
     for (ColumnData<T> column in widget.columns) {
-      final width = column.fixedWidthPx ?? (available / unconstrainedCount);
+      double width = column.fixedWidthPx;
+      if (width == null) {
+        width = available / unconstrainedCount;
+        if (column.minWidthPx != null) {
+          width += column.minWidthPx;
```

Comment:
should this be `width = math.max(width, column.minWidthPx)`? Otherwise we could be adding minWidthPx to the divided space for unconstrained columns.

Reply:
Good point. I was afraid that would lead to edge case issues calculating the available width but it is no worse than what I had for the edge cases.

Reply:
actually I am wrong. It is tricky. There is an extra edge case if we do it that way. The problem is when some columns are at their minimum constraints and some aren't. The option you suggested will result in a bug where sometimes we don't actually use the available width or we use too much width. We'll need a 3rd pass to do it correctly. Shouldn't be hard, let me go ahead and add that.